manim/grpc/impl/frame_server_impl.py

```diff
@@ -10,250 +10,279 @@
+            requested_scene.update_to_time(animation_offset)
 
-        if selected_scene.animations:
-            # This is a call to play().
-            selected_scene.update_animation_to_time(request.animation_offset)
-            selected_scene.update_frame(
-                selected_scene.moving_mobjects,
-                selected_scene.static_image,
+            # Serialize the scene's mobjects.
+            mobjects = extract_mobject_family_members(
+                requested_scene.mobjects, only_those_with_points=True
             )
-            serialized_mobject_list, duration = selected_scene.add_frame(
-                selected_scene.renderer.get_frame()
-            )
-            resp = list_to_frame_response(
-                selected_scene, duration, serialized_mobject_list
+            serialized_mobjects = [
+                serialize_mobject(mobject)
+                for mobject in mobjects
+                if not isinstance(mobject, ValueTracker)
+            ]
+
+            resp = frameserver_pb2.FrameResponse(
+                mobjects=serialized_mobjects,
+                frame_pending=False,
+                animation_finished=False,
+                scene_finished=scene_finished
+                or request.preview_mode
+                == frameserver_pb2.FrameRequest.PreviewMode.IMAGE,
+                duration=requested_scene.duration,
+                animations=map(
+                    lambda anim: anim.__class__.__name__, requested_scene.animations
+                ),
+                animation_index=requested_scene_index,
+                animation_offset=animation_offset,
             )
             return resp
-        else:
-            # This is a call to wait().
-            if selected_scene.should_update_mobjects():
-                # TODO, be smart about setting a static image
-                # the same way Scene.play does
-                selected_scene.update_animation_to_time(time)
-                selected_scene.update_frame()
-                serialized_mobject_list, duration = selected_scene.add_frame(
-                    selected_scene.get_frame()
-                )
-                frame_response = list_to_frame_response(
-                    selected_scene, duration, serialized_mobject_list
+        except Exception as e:
+            traceback.print_exc()
+
+    def FetchSceneData(self, request, context):
+        try:
+            return frameserver_pb2.FetchSceneDataResponse(
+                scene=frameserver_pb2.Scene(
+                    name=str(self.scene),
+                    animations=[
+                        frameserver_pb2.Animation(
+                            name=animations_to_name(scene.animations),
+                            duration=scene.duration,
+                        )
+                        for scene in self.keyframes
+                    ],
+                ),
+            )
+        except Exception as e:
+            traceback.print_exc()
```

Comment:
Here also. Is this any different than printing `e`?

Reply:
When I just print(e) the location of the error isn't printed.